### DISCUSSION TITLE
Docker for PROD/LIVE environment?

### DISCUSSION THREAD

**Discussion by cmann90** [pimcore/pimcore#12899] [Q&A / Support] [answered]
Is Docker a good alternative to VM's for Pimcore?

We have been using Docker for almost two years for local development as well as our BETA, TEST and PROD/LIVE environment. The environments run very stable but lately there have been doubts if the performance on a VM would be better or if there are any other advantages to run Pimcore on a VM instead of Docker.

The main reason we started off with Docker is that we want to have multiple Pimcore environments with the same configuration. Also our knowledge to install and maintain a Linux OS is currently very limited. 

Thanks for any shared experiences and suggestions.

@dpfaffenbauer I found an older reply of yours here: https://talk.pimcore.org/t/multi-environment-dev-prod/1214
> I have 3 environments: DEV (local), STAGING (pre-prod) and PROD.
> - My local environment is setup using docker
> - The STAGING and PROD are usually on different Servers but with the same setup (in most cases cPanel/WHM or Plesk)

If you happen to read this Thank You for mentioning Deployer and everything you shared with the community. We didn't know that tool and will definitely take a look at it.

Regarding your old reply. I am not sure if you actually use Docker for your STAGING and PROD environments or if you only meant that STAGING and PROD are configured the same. Would be nice to have your opinion on this question as you basically maintain the Docker images for Pimcore.

**Comment by dpfaffenbauer** [comment, marked as answer]
we run Pimcore X in Kubernetes with Containerd (docker) now...

**Reply by cmann90** [reply]
Basically answers my question already. Thank you very much. Input from others still very much welcome.

**Comment by cmann90** [comment]
**Update**
Dominik Pfaffenbauer who already provided an answer to this question posted an article about Pimcore with Kubernetes on the official Pimcore blog:
https://pimcore.com/en/resources/blog/kubernetes-with-pimcore_a271065

The blog article also includes a link to an example git repository:
https://github.com/cors-gmbh/pimcore-k8s-demo

**Reply by dpfaffenbauer** [reply]
Cheers mate
